src/ast/query.rs

```diff
@@ -835,6 +843,24 @@ impl fmt::Display for TableAlias {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum TableVersion {
+    Timestamp(String),
```

Comment:
I looked at MsSQL and it seems to allow several different phrases like `BETWEEN` and `CONTAINED`

https://learn.microsoft.com/en-us/sql/relational-databases/tables/querying-data-in-a-system-versioned-temporal-table?view=sql-server-ver16

I don't think we need to add support for any of this syntax now in this PR, but your choice of `enum` to represent this structure I think is prescient

Reply:
Good idea, will revise it like that.

Reply:
Ok, I've renamed `AsOf` to `ForSystemTimeAsOf`, and made it wrap a generic `Expr` instead of a `String`. Arguably this makes the clause more versatile then is supported by the dialects, but that's probably for the better.